src/js/hooks/productSupplier/form/useProductSupplierForm.js

```diff
@@ -0,0 +1,150 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { useParams } from 'react-router-dom';
+import { z } from 'zod';
+
+import productSupplierApi from 'api/services/ProductSupplierApi';
+import { omitEmptyValues } from 'utils/form-values-utils';
+
+const useProductSupplierForm = () => {
+  const validationSchema = z
+    .object({
+      id: z.string().optional(),
+      code: z.string().optional(),
+      product: z.object({
+        id: z.string(),
+        value: z.string(),
+        label: z.string(),
+      }, {
+        invalid_type_error: 'Product is required',
+        required_error: 'Product is required',
+      }).required(),
+      legacyCode: z
+        .string()
+        .optional(),
+      supplier: z.object({
+        id: z.string(),
+        value: z.string(),
+        label: z.string(),
+      }, {
+        invalid_type_error: 'Supplier is required',
+        required_error: 'Supplier is required',
+      }).required(),
+      supplierCode: z
+        .string({ required_error: 'Supplier code is required' })
+        .min(1, 'Supplier code is required')
+        .max(255, 'Max length of supplier code is 255'),
+      name: z
+        .string({ required_error: 'Supplier Product Name is required' })
+        .min(1, 'Supplier Product Name is required')
+        .max(255, 'Max length of supplier product name is 255'),
+      active: z.boolean(),
+      manufacturer: z.object({
+        id: z.string(),
+        value: z.string(),
+        label: z.string(),
+      }).optional(),
+      ratingTypeCode: z.object({
+        id: z.string(),
+        value: z.string(),
+        label: z.string(),
+      }).optional(),
+      manufacturerCode: z
+        .string()
+        .optional(),
+      brandName: z
+        .string()
+        .optional(),
+    });
```

Comment:
What do you think about moving this schema into a separate file instead of having it in a hook? 
cc @alannadolny

Reply:
Sounds good and I was even thinking about it, but didn’t see that much value of it, but if you guys prefer it, I will move it to a separate file, since it doesn’t depend on React lifecycle.

Reply:
if you can then please move it, I think this will improve readability of this schema

Reply:
That sounds good, but it doesn't look wrong to me at this moment.

Reply:
Plus these should be translatable, but that can be left for later.

Reply:
oh, yeah, I again forgot about it - so this changes the situation - to have them translatable, I need to use it in the hook, hence I'd need `translate()` method from the `useDispatch()`.

Reply:
Unless I'd suggest creating a separate hook for the validation schema -> `useProductSupplierValidation()` and put there the validation schema.
but I also agree let's do it "by the way"